planck-cljs/src/planck/io.cljs

```diff
@@ -346,13 +346,13 @@
 (defn ^boolean directory?
   "Checks if dir is a directory."
   [dir]
-  (js/PLANCK_IS_DIRECTORY (:path (as-file dir))))
+  (boolean (js/PLANCK_IS_DIRECTORY (:path (as-file dir)))))
```

Comment:
@mfikes one of the tests I added for `directory?` failed since it returned `nil` when given `nil` as an parameter.  This change fixes that.

Reply:
Cool. That could be taken care of as a separate ticket. (Unrelated to the new predicates being added.)

Reply:
It certainly could, though doing so would require removal of some/all of the `directory?` unit tests in this PR.  What would you prefer?

Reply:
Seems fine as is, as all of these predicates can be thought of as going together.

Reply:
Oh, looking at this one again... nil is not a valid argument, according to the spec. No need to add the cast to boolean as the function will behave properly when passed valid arguments.

Reply:
The [unit tests were quite happy having `nil` passed in](https://github.com/planck-repl/planck/pull/864/files#diff-742e4644324187fb2e256653a20bd1efR40) but the tests failed until I added this coercion.

Again, I'm brand new to Clojure specs and unclear how argument specifications are verified upon function invocation.

Reply:
Specs are verified if `clojure.spec.test.alpha/instrument` is invoked.